cryptolib/attestation.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cryptolib
+
+type (
+	// KeyMode stores the type of the Attestation.
+	KeyMode int
```

Comment:
Key Mode is entirely determined by Signature Algorithm, I'd remove it and update Signature Algorithm to be pkix_*, pgp_* etc

Reply:
Anthony and I discussed this earlier today and  thought of KeyMode having a SignatureAlgorithm field. Would you be opposed to this rather than adding the KeyMode as a prefix in the SignatureAlgorithm?

Reply:
I'm note sure I follow. Do you mean creating a type which contains the information in KeyMode and SigningAlgoritm? That sounds reasonable to me.

Also small nit: SigningAlgoritm-> SignatureAlgorithm becaues it is used in verification as well.

Reply:
Done. We opted for the prefix solution that you suggested. We also changed the name to `SignatureAlgorithm`.